### PR TITLE
Adding Subscriptions

### PR DESCRIPTION
**Pull request by tlemburg**

### CODE REVIEW

File: src/UNL/UCBCN/Calendar/SubscriptionHasCalendar.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    public $id;                              // int(10)  not_null primary_key unsigned auto_increment
+    public $subscription_id;                // int(10)  not_null multiple_key unsigned
+    public $calendar_id;                     // int(10)  not_null multiple_key unsigned
+
+    public static function getTable()
+    {
+        return 'subscription_has_calendar';
+    }
+
+    function table()
```
Comment (mfairchild365) [inline src/UNL/UCBCN/Calendar/SubscriptionHasCalendar.php]: I don't think our active record implementation requires this method.

File: src/UNL/UCBCN/Calendar/SubscriptionHasCalendar.php [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    function keys()
+    {
+        return array(
+            'id',
+        );
+    }
+    
+    function sequenceKey()
```
Comment (mfairchild365) [inline src/UNL/UCBCN/Calendar/SubscriptionHasCalendar.php]: Nor this

File: src/UNL/UCBCN/Calendar/SubscriptionHasCalendar.php [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    {
+        return array('id',true);
+    }
+    
+    function links()
```
Comment (mfairchild365) [inline src/UNL/UCBCN/Calendar/SubscriptionHasCalendar.php]: Nor this

File: src/UNL/UCBCN/Events.php [hunk 4]
```diff
@@ -26,7 +26,21 @@ public function getDefaultOptions() {
...
+                        calendar_has_event.status != "pending" AND event.approvedforcirculation = 1 AND (';
+
+            # add the calendars requested
+            $sql .= implode(' OR ', array_map(function($calendar_id) {return 'calendar_has_event.calendar_id = ' . (int)$calendar_id;}, $this->options['subscription_calendars']));
+
+            # exclude the calendar that is doing the subscribing (exclude events that it already contains)
+            $sql .= ") AND event.id NOT IN (SELECT DISTINCT event.id FROM event, calendar_has_event AS c2 WHERE c2.calendar_id = " . 
+                $this->options['subscription_calendar'] . " AND c2.event_id = event.id);";
+
+error_log($sql);
```
Comment (mfairchild365) [inline src/UNL/UCBCN/Events.php]: debug statement

**Comment by mfairchild365** [comment]
Were you able to create a migration script?

**Comment by tlemburg** [comment]
Oh crap. There is one started, but I have not yet finished it.

**Pull request merged**